test/endecode_test.c

```diff
@@ -24,6 +24,71 @@
 
 #include "testutil.h"
 
+static EVP_PKEY *get_dh_from_pg_bn(OSSL_LIB_CTX *libctx, BIGNUM *p, BIGNUM *g)
+{
+    EVP_PKEY_CTX *pctx = EVP_PKEY_CTX_new_from_name(libctx, "DH", NULL);
```

Comment:
We seem to have forgotten the `type` argument that was originally passed to `get_dh_from_pg()`

Reply:
Oops, right, will add...

Reply:
Yeah, I'll also move the helper functions that are in common with `ssltest_old.c` to `testutil/`.
I can do this evening.

Reply:
Done.

Reply:
I also found that DHX keys required other data and a generalization of the parameter generation; done as well.